api/src/test/java/org/pmiops/workbench/db/dao/UserServiceTest.java

```diff
@@ -192,54 +189,78 @@ public void testSyncComplianceTrainingStatusV2() throws Exception {
+    long issued = fakeClock.instant().getEpochSecond() - 10;
+    BadgeDetailsV2 retBadge = new BadgeDetailsV2().lastissued(issued).valid(true);
 
-    Map<String, BadgeDetailsV2> userBadgesByName = new HashMap<>();
-    userBadgesByName.put(mockComplianceService.getResearchEthicsTrainingField(), retBadge);
+    Map<BadgeName, BadgeDetailsV2> userBadgesByName = new HashMap<>();
+    userBadgesByName.put(BadgeName.REGISTERED_TIER_TRAINING, retBadge);
 
     when(mockComplianceService.getUserBadgesByBadgeName(USERNAME)).thenReturn(userBadgesByName);
 
     userService.syncComplianceTrainingStatusV2();
 
-    // The user should be updated in the database with a non-empty completion and expiration time.
+    // The user should be updated in the database with a non-empty completion time.
     DbUser user = userDao.findUserByUsername(USERNAME);
-    assertThat(user.getComplianceTrainingExpirationTime())
-        .isEqualTo(Timestamp.from(Instant.ofEpochSecond(expiry)));
     assertModuleCompletionEqual(
         AccessModuleName.RT_COMPLIANCE_TRAINING, user, Timestamp.from(START_INSTANT));
 
     // Deprecate the old training.
-    long newExpiry = expiry - 1;
-    retBadge.setDateexpire(newExpiry);
     retBadge.setValid(false);
 
     // Completion timestamp should be wiped out by the expiry timestamp passing.
     userService.syncComplianceTrainingStatusV2();
     assertModuleCompletionEqual(AccessModuleName.RT_COMPLIANCE_TRAINING, user, null);
 
     // The user does a new training.
-    long newerExpiry = expiry + 1;
-    retBadge.setDateexpire(newerExpiry);
-    retBadge.setValid(true);
+    retBadge.lastissued(issued + 5).valid(true);
 
     // Completion and expiry timestamp should be updated.
     userService.syncComplianceTrainingStatusV2();
-    assertThat(user.getComplianceTrainingExpirationTime())
-        .isEqualTo(Timestamp.from(Instant.ofEpochSecond(newerExpiry)));
     assertModuleCompletionEqual(
         AccessModuleName.RT_COMPLIANCE_TRAINING, user, Timestamp.from(START_INSTANT));
 
-    // A global expiration is set.
-    long globalExpiry = expiry - 1000;
-    retBadge.setGlobalexpiration(globalExpiry);
-    retBadge.setValid(false);
+    // Time passes, user renews training
+    retBadge.lastissued(fakeClock.instant().getEpochSecond() + 1);
+    fakeClock.increment(5000);
 
-    // Completion timestamp should be wiped out by the globalexpiry timestamp passing.
+    // Completion should be updated to the current time.
     userService.syncComplianceTrainingStatusV2();
-    assertModuleCompletionEqual(AccessModuleName.RT_COMPLIANCE_TRAINING, user, null);
+    assertModuleCompletionEqual(
+        AccessModuleName.RT_COMPLIANCE_TRAINING, user, Timestamp.from(fakeClock.instant()));
+  }
+
+  @Test
+  public void testUpdateComplianceTrainingStatusV2_controlled() throws Exception {
+    long issued = fakeClock.instant().getEpochSecond() - 10;
+    BadgeDetailsV2 ctBadge = new BadgeDetailsV2().lastissued(issued).valid(true);
```

Comment:
nit: it's awkward to specify the rt and ct badges in different ways

Reply:
Slightly awkward, but my preference is to minimize local state / variables where possible (within reason) as it generally reduces load on the reader. ctBadge needs to be in a variable as it's getting mutated later - rtBadge is just incidental. Let me know if you feel strongly on this point.

Reply:
ok np